### DISCUSSION TITLE
"Server Restarting Soon" Alert for MP

### DISCUSSION THREAD

**Discussion by RuinTD** [ppy/osu#16134] [Ideas] [answered]
![image](https://user-images.githubusercontent.com/1149870/146589064-d9c54cf3-e417-4c03-9aae-cfbb9dfb494c.jpeg)
In response to this post in the lazer changelog, I think there should be a feature to alert users in lobbies that the server is rebooting soon (maybe within 10 minutes). Could be helpful for both updates and regular server maintenance. It could be an overlay that has the chat at the bottom (so people in game can still talk) and saying something like:

"The server is rebooting soon to prepare for a game update. What do you want to do?”
“Play it safe (Leave room)”
"Just one more game (Keep playing)"

This overlay could also be used to warn users when they try to create new rooms. 

"Play it safe (Cancel)"
"Just a quick game (Create room)"

**Comment by peppy** [comment, marked as answer]
the plan is to do rolling deploys, so you won't even noticed it happening. all existing rooms will remain running on the previous deployment. you'll see this applied probably within the first half of next year.
